tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h

```diff
@@ -0,0 +1,482 @@
+
+  bool setDynamicRange(float min, float max) {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->setDynamicRange(min, max);
+      case TensorType::kSIMPLE: return simple_tensor_->setDynamicRange(min, max);
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+
+  bool dynamicRangeIsSet() const { 
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->dynamicRangeIsSet();
+      case TensorType::kSIMPLE: return simple_tensor_->dynamicRangeIsSet();
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+
+  void resetDynamicRange() {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->resetDynamicRange();
+      case TensorType::kSIMPLE: return simple_tensor_->resetDynamicRange();
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+  float getDynamicRangeMin() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->getDynamicRangeMin();
+      case TensorType::kSIMPLE: return simple_tensor_->getDynamicRangeMin();
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+
+  float getDynamicRangeMax() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->getDynamicRangeMax();
+      case TensorType::kSIMPLE: return simple_tensor_->getDynamicRangeMax();
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+#if IS_TRT_VERSION_GE(5, 0, 0, 0) && !IS_TRT_VERSION_GE(8, 0, 0, 0)
+  float getDynamicRange() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->getDynamicRange();
+      case TensorType::kSIMPLE: return simple_tensor_->getDynamicRange();
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+#endif
+  void setAllowedFormats(nvinfer1::TensorFormats formats)
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->setAllowedFormats(formats);
+      case TensorType::kSIMPLE: return simple_tensor_->setAllowedFormats(formats);
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+
+  nvinfer1::TensorFormats getAllowedFormats() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->getAllowedFormats();
+      case TensorType::kSIMPLE: return simple_tensor_->getAllowedFormats();
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+
+  bool isShapeTensor() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->isShapeTensor();
+      case TensorType::kSIMPLE: return simple_tensor_->isShapeTensor();
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+
+  bool isExecutionTensor() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->isExecutionTensor();
+      case TensorType::kSIMPLE: return simple_tensor_->isExecutionTensor();
+    }
+    assert(0 && "Unsupported itensor_ type");
+  }
+
+private:
+  bool validate() const
+  {
+    return trt_tensor_ && !simple_tensor_ || !trt_tensor_ && simple_tensor_;
```

Comment:
Need to add parenthesis, ( trt_tensor_ && !simple_tensor_&&  ) || (!trt_tensor_ && simple_tensor_)  to avoid this warning https://stackoverflow.com/questions/16939888/why-does-clang-warn-within

Reply:
Good point, added now